src/System Application/App/Email/src/Email/EmailDispatcher.Codeunit.al

```diff
@@ -42,20 +44,38 @@ codeunit 8888 "Email Dispatcher"
         // -----------
 
         Rec.LockTable(true);
-        if EmailRateLimitImpl.IsRateLimitExceeded(Rec."Account Id", Rec.Connector, Rec."Send From", RateLimitDuration) then
+        if EmailRateLimitImpl.IsRateLimitExceeded(Rec."Account Id", Rec.Connector, Rec."Send From", RateLimitDuration)
+        or (GetEmailOutboxCurrentProcessingCount() > EmailRateLimitImpl.GetConcurrencyLimit(Rec."Account Id", Rec.Connector, Rec."Send From")) then
```

Comment:
Should this logic be part of `IsRateLimitExeeded`?

If not, consider making a new procedure with a well-defined name